### ISSUE TITLE
Suggestion: support jpeg output too

### ISSUE THREAD

**Issue by ktei** [cytoscape/cytoscape.js#926]
It'd be better if cytoscape supports jpeg image export as well, for example:
```javascript
  CanvasRenderer.prototype.jpg = function( options ){
    return this.bufferCanvasImage( options ).toDataURL('image/jpeg');
  };
```

**Comment by maxkfranz** [comment]
Because the graphs are essentially line drawings and JPEG introduces artefacts (i.e. blurriness), I didn't want to add that.

Maybe it should be a simple extension?  The problem is JPEG is a more ubiquitous format than PNG, so if both were included by default -- a lot of people would make the mistake of just using JPEG and then being dissatisfied with the results.  

Is the concern only with filesize?

**Comment by ktei** [comment]
@maxkfranz 
One of the requirements of my current project is to export the graph as PDF. However, if (this doesn't not happen often) the graph is very large, then when export, if I choose a big scale such as 5, the buffer canvas size (width/height) could be more than 20000, which results in a LOT base64 characters, which exceeds the `toDataURL` limit. If I use `toDataURL('image/jpeg')` then the number of characters can be dramatically reduced.

I know that I should not use scale of 5 in this case, so I wrote a loop, in which I kept reducing the scale until the buffer canvas size is less than 10000. You may ask why I need such high scale, that's because I tried using scale of 2.5 but the output image on PDF (even in PNG format) is not clear enough (blurry).

So basically, the `toDataURL` limit is a concern here. However, by using JPEG, I can use high scale and won't exceed the limit too.

I've changed my forked cytoscape project to output JPEG, so I was wondering maybe the other users could have similar problem as mine.

**Comment by maxkfranz** [comment]
That makes sense.  I didn't know about that limitation.  Maybe a strong warning in the docs would be enough then.

**Comment by maxkfranz** [comment]
I'm planning on releasing on Monday, so I just sneaked this in.
